qiskit/quantum_info/operators/symplectic/stabilizer_circuit.py

```diff
@@ -0,0 +1,189 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017--2023
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Stabilizer to circuit function
+"""
+from __future__ import annotations
+
+from qiskit.exceptions import QiskitError
+
+from qiskit.circuit import QuantumCircuit
+
+from .clifford import Clifford
+from .pauli import Pauli
+
+
+def add_sign(stabilizer: str) -> str:
+    """
+    Add a sign to stabilizer if it is missing.
+    :param stabilizer: stabilizer string
+    :return: stabilizer string with sign
+    """
+    if stabilizer[0] not in ["+", "-"]:
+        return "+" + stabilizer
+    return stabilizer
+
+
+def drop_sign(stabilizer: str) -> str:
+    """
+    Drop sign from stabilizer if it is present.
+    :param stabilizer: stabilizer string
+    :return: stabilizer string without sign
+    """
+    if stabilizer[0] not in ["+", "-"]:
+        return stabilizer
+    if stabilizer[1] == "i":
+        return stabilizer[2:]
+    return stabilizer[1:]
+
+
+def check_stabilizers_commutator(s_1: str, s_2: str) -> bool:
+    """
+    Check if two stabilizers commute.
+    :param s_1: stabilizer string
+    :param s_2: stabilizer string
+    :return: True if stabilizers commute, False otherwise
+    """
+    prod = 1
+    for o1, o2 in zip(drop_sign(s_1), drop_sign(s_2)):
+        if o1 == "I" or o2 == "I":
+            continue
+        if o1 != o2:
+            prod *= -1
+    return prod == 1
+
+
+def apply_circuit_on_stabilizer(stabilizer: str, circuit: QuantumCircuit) -> str:
+    """
+    Given a stabilizer string and a circuit, conjugate the circuit on the stabilizer.
+    :param stabilizer: stabilizer string
+    :param circuit: Clifford circuit to apply
+    :return: stabilizer string after conjugation
+    """
+    cliff = Clifford(circuit)
+    stab_operator = Pauli(stabilizer)
+    pauli_conjugated = stab_operator.evolve(cliff, frame="s")
+    return pauli_conjugated.to_label()
+
+
+def stabilizer_to_circuit(
```

Comment:
a lot of the API in `qiskit.quantum_info` is like `object.from(...)`.
so should this API be accessible to users like `Clifford.from_stabilizers(...)`
(where is the current API for building Clifford circuits from stabilizers and destabilizers?)

Reply:
Indeed, I agree that it should be a method of a `quantum_info` class and not a standalone function. 
However, I would suggest that if will be a method of the `StabilizerState` class and not the `Clifford` class, see our comment here: https://github.com/Qiskit/qiskit/issues/11478#issuecomment-1875120276

Reply:
I'm not sure about this. The `from(...)` syntax kind of suggests that there is a unique way to transform input to output, which is not the case even with a fully specified set of stabilizers (there are multiple non-equivalent circuits that transform |0> to stabilizer state).